pkg/manager/member/tidbcluster_status_manager.go

```diff
@@ -14,37 +14,180 @@
 package member
 
 import (
+	"encoding/json"
+	"fmt"
+
 	"github.com/pingcap/tidb-operator/pkg/apis/pingcap/v1alpha1"
 	"github.com/pingcap/tidb-operator/pkg/client/clientset/versioned"
+	"github.com/pingcap/tidb-operator/pkg/pdapi"
 	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
+)
+
+const (
+	prometheusEtcdKey = "/topology/prometheus"
+	grafanaEtcdKey    = "/topology/grafana"
+	//TODO support AlertManager, move to UCP
+	alertManagerEtcdKey = "/topology/alertmanager"
 )
 
 type TidbClusterStatusManager struct {
-	cli versioned.Interface
+	cli       versioned.Interface
+	pdControl pdapi.PDControlInterface
 }
 
-func NewTidbClusterStatusManager(cli versioned.Interface) *TidbClusterStatusManager {
+func NewTidbClusterStatusManager(kubeCli kubernetes.Interface, cli versioned.Interface) *TidbClusterStatusManager {
 	return &TidbClusterStatusManager{
-		cli: cli,
+		cli:       cli,
+		pdControl: pdapi.NewDefaultPDControl(kubeCli),
 	}
 }
 
 func (tcsm *TidbClusterStatusManager) Sync(tc *v1alpha1.TidbCluster) error {
+	return tcsm.syncTidbMonitorRefAndKey(tc)
+}
+
+func (tcsm *TidbClusterStatusManager) syncTidbMonitorRefAndKey(tc *v1alpha1.TidbCluster) error {
+	tm, err := tcsm.syncTidbMonitorRef(tc)
+	if err != nil {
+		return err
+	}
+	return tcsm.syncDashboardMetricStorage(tc, tm)
```

Comment:
Status manager should focus on the status update, can we move the etcd key operation to a separate function out of the status manager?

Reply:
Currently, i think it is not necessary as it seems we would only solve the `MonitorRef` and `AutoSclaerRef` (in future) here. If we movet the etcd operation out of statusManager, we will try to fetch `TidbMonitor` again in its scope.